### PR TITLE
Remove from network

### PR DESCRIPTION
**Pull request by frankfu03**

### CODE REVIEW

File: gws-core/src/test/java/au/gov/ga/geodesy/port/adapter/rest/CorsSiteEndpointITest.java [hunk 1]
```diff
@@ -83,4 +83,38 @@ public void checkNetworkTenancy() throws Exception {
...
+    @Rollback(false)
+    public void checkNetworkTenancyAfterRemove() throws Exception {
+        List<NetworkTenancy> networkTenancies = given()
+            .when()
+            .get("/corsSites/search/findByFourCharacterId?id=ALIC")
+            .then()
+            .statusCode(HttpStatus.OK.value())
+            .extract().jsonPath().getList("networkTenancies", NetworkTenancy.class);
+
+        assertThat(networkTenancies.size(), is(0));
```
Comment (lbodor) [inline gws-core/src/test/java/au/gov/ga/geodesy/port/adapter/rest/CorsSiteEndpointITest.java]: assertThat(networkTenancies).isEmpty()

**Comment by lbodor** [comment]
I recently merged some PRs onto master, which should help with the failling test suite.

File: gws-core/pom.xml [hunk 2]
```diff
@@ -117,7 +117,7 @@
         <dependency>
             <groupId>au.gov.ga.geodesy</groupId>
             <artifactId>igssitelog-bindings</artifactId>
-            <version>0.1.0</version>
```
[the remark this answers is not in the document]
Comment (frankfu03) [reply gws-core/pom.xml]: It failed in local build. The version should be updated as shown in igssitelog-java-bindings/pom.xml: 

```
<groupId>au.gov.ga.geodesy</groupId>
    <artifactId>igssitelog-bindings</artifactId>
    <packaging>jar</packaging>
    <name>IGS Site Log Bindings</name>
    <version>0.1.1-SNAPSHOT</version>
```

File: gws-core/src/main/java/au/gov/ga/geodesy/support/spring/GeodesyRepositoryRestMvcConfig.java [hunk 3]
```diff
@@ -70,6 +70,14 @@ public RootResourceProcessor getRootResourceProcessor() {
                     .slash("addToNetwork");
 
                 resource.add(link.withRel("addToNetwork"));
+
+                LinkBuilder link2 = configuration.entityLinks().linkForSingleResource(
```
[the remark this answers is not in the document]
Comment (frankfu03) [reply gws-core/src/main/java/au/gov/ga/geodesy/support/spring/GeodesyRepositoryRestMvcConfig.java]: Done.

File: gws-core/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (frankfu03) [reply gws-core/pom.xml]: I pulled the code from **next** branch. The version is 0.1.0 in `master ` branch. I'll revert it to 0.1.0.

File: gws-core/src/main/java/au/gov/ga/geodesy/domain/model/CorsSite.java [hunk 4]
```diff
@@ -120,4 +122,24 @@ public void setSiteStatus(String siteStatus) {
...
+        if (index != -1) {
+            this.networkTenancies.remove(index);
+            return Stream.of(new CorsSiteRemovedFromNetwork(this.getId(), command.getNetworkId(), command.getEffectiveFrom()));
+        } else {
+            return Stream.empty();
+        }
+    }
+
+    private int findNetworkTenancyIndexByNetworkId(Integer networkId) {
+        for (int i = 0; i < this.networkTenancies.size(); i ++) {
```
[the remark this answers is not in the document]
Comment (frankfu03) [reply gws-core/src/main/java/au/gov/ga/geodesy/domain/model/CorsSite.java]: It is done. better now, thanks.

File: gws-core/src/main/resources/db/geodesy-database-changelog.xml [hunk 5]
```diff
@@ -63,4 +63,5 @@
     <include file="db/init-cors-networks-versions.sql"/>
     <include file="db/rename-setup-name-column.xml"/>
     <include file="db/rename-cors-setup-type.sql"/>
+    <include file="db/cors-site-removed-from-network.xml"/>
```
[the remark this answers is not in the document]
Comment (frankfu03) [reply gws-core/src/main/resources/db/geodesy-database-changelog.xml]: OK, done.

File: gws-system-test/src/main/bash/remove-site-from-network.sh [hunk 6]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+set -e
```
[the remark this answers is not in the document]
Comment (frankfu03) [reply gws-system-test/src/main/bash/remove-site-from-network.sh]: Done.

File: gws-system-test/src/main/bash/remove-site-from-network.sh [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+clientId=GnssSiteManager
+clientPassword=
+
+username=
+password=
+
+jwt=$(curl -s --user ${clientId}:${clientPassword} --data "grant_type=password&username=${username}&password=${password}&scope=openid profile" ${openam}/oauth2/access_token?realm=/ | jq .id_token | tr -d '"')
+
+function removeSiteFromNetwork {
+    removeFromNetwork=$(curl -s ${gws}/corsSites?fourCharacterId=${1} | jq ._embedded.corsSites[0]._links.removeFromNetwork.href | tr -d '"')
```
[the remark this answers is not in the document]
Comment (frankfu03) [reply gws-system-test/src/main/bash/remove-site-from-network.sh]: Yeah, done.

File: gws-core/src/test/java/au/gov/ga/geodesy/port/adapter/rest/CorsSiteEndpointITest.java (hunk 1 above)
Comment (frankfu03) [reply gws-core/src/test/java/au/gov/ga/geodesy/port/adapter/rest/CorsSiteEndpointITest.java]: Good, it is done.

Comment (frankfu03) [reply gws-core/src/test/java/au/gov/ga/geodesy/port/adapter/rest/CorsSiteEndpointITest.java]: Changed to `assertThat(networkTenancies).isEmpty();` but got a compilation error:

_no suitable method found for assertThat(java.util.List<au.gov.ga.geodesy.domain.model.NetworkTenancy>)_

So I used `assertThat(networkTenancies, is(empty()));`

_In addition, `org.junit.Assert.assertTrue(networkTenancies.isEmpty());` also works, but it seems that we prefer **hamcrest** to **junit** (only two test classes used junit: GnssStationRepositoryTest and StationRepositoryTest)_

Comment (lbodor) [reply gws-core/src/test/java/au/gov/ga/geodesy/port/adapter/rest/CorsSiteEndpointITest.java]: That's correct. My suggestion is for assertj framework that we use in GNSS data repo, I forgot that GWS used junit.

**Comment by frankfu03** [comment]
I rebased it from master. it is now running checking on travis-ci.

**Pull request merged**